### PR TITLE
ADM - 436 Add calculate logic for lead time.

### PR DESCRIPTION
**Pull request by WSSsssss33**
## Summary
Add calculate logic for lead time.
...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by guzhongren — CHANGES_REQUESTED** [review]
In [your commit](https://github.com/au-heartbeat/Heartbeat/pull/439/commits/b0cd685a795b7e8932b9de46852dea3189b47e52), tokens are still there, need you to rewrite your code on this commit

**Comment by WSSsssss33** [comment]
> In [your commit](https://github.com/au-heartbeat/Heartbeat/pull/439/commits/b0cd685a795b7e8932b9de46852dea3189b47e52), tokens are still there, need you to rewrite your code on this commit

Token has been faked in this  [commit](https://github.com/au-heartbeat/Heartbeat/pull/439/commits/d84dd06aa80435f8ab355dd97ad081a3eb946652)

File: backend/src/main/java/heartbeat/service/report/LeadTimeForChangesCalculator.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		int pipelineCount = pipelineLeadTime.size();
+		List<LeadTimeForChangesOfPipelines> leadTimeForChangesOfPipelines = new ArrayList<>();
+		AvgLeadTimeForChanges avgLeadTimeForChanges = new AvgLeadTimeForChanges();
+
+		if (pipelineLeadTime.isEmpty()) {
+			return new LeadTimeForChanges(leadTimeForChangesOfPipelines, avgLeadTimeForChanges);
+		}
+
+		List<HashMap<String, Double>> avgDelayTimeMapList = pipelineLeadTime.stream().map(item -> {
+			int times = item.getLeadTimes().size();
```
Comment (TingLiMio) [inline backend/src/main/java/heartbeat/service/report/LeadTimeForChangesCalculator.java:32]: Maybe you should do `item.getLeadTimes().isEmpty()` before `item.getLeadTimes().size()`.

File: backend/src/main/java/heartbeat/service/report/GenerateReporterService.java [hunk 2]
```diff
@@ -129,21 +148,33 @@ private void fetchDataFromKanban(GenerateReportRequest request) {
...
+		fetchBuildKiteData(request.getStartTime(), request.getEndTime(), request.getCodebaseSetting().getLeadTime(),
+				request.getBuildKiteSetting().getToken());
+		Map<String, String> repoMap = getRepoMap(request.getCodebaseSetting());
+		this.leadTimes = gitHubService
+			.fetchPipelinesLeadTime(this.deployTimesList, repoMap, request.getCodebaseSetting().getToken())
+			.join();
 	}
 
 	private void fetchBuildKiteData(GenerateReportRequest request) {
+		fetchBuildKiteData(request.getStartTime(), request.getEndTime(),
```
Comment (TingLiMio) [inline backend/src/main/java/heartbeat/service/report/GenerateReporterService.java:161]: The two method names are the same, is it not easy to distinguish?  `fetchBuildKiteData `

**Review by guzhongren — APPROVED** [review]

File: backend/src/main/java/heartbeat/service/report/LeadTimeForChangesCalculator.java (hunk 1 above)
Comment (WSSsssss33) [reply backend/src/main/java/heartbeat/service/report/LeadTimeForChangesCalculator.java:32]: will be fix in next commit

File: backend/src/main/java/heartbeat/service/report/GenerateReporterService.java (hunk 2 above)
Comment (WSSsssss33) [reply backend/src/main/java/heartbeat/service/report/GenerateReporterService.java:161]: No confusion, their functions are the same, but the parameters are different

**Pull request merged**